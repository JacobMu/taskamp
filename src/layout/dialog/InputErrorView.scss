@import "~@blueprintjs/core/lib/scss/variables";

.formGroupError {
  margin: 0.5rem 0;
  background-color: $vermilion5;
  border: 2px $vermilion3 solid;
  color: $white;
}
