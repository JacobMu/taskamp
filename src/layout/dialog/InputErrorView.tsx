import React from "react";
import { Card } from "@blueprintjs/core";

import './InputErrorView.scss';

interface Props {
  i18n?: string;
}

export const InputErrorView: React.FC<Props> = ({ i18n = 'app.UNKNOWN_ERROR' }) => (
  <Card className="formGroupError">{i18n}</Card>
);
