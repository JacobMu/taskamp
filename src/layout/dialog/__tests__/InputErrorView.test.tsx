import React from "react";
import { shallow } from "enzyme";

import { InputErrorView } from "../InputErrorView";

describe("<InputErrorView />", () => {
  it("renders input error card with message", () => {
    const wrapper = shallow(<InputErrorView i18n="errorMsg" />);

    expect(wrapper).toMatchSnapshot();
  });
});
