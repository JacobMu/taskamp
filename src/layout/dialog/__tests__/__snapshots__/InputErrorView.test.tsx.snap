// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputErrorView /> renders input error card with message 1`] = `
<Blueprint3.Card
  className="formGroupError"
  elevation={0}
  interactive={false}
>
  errorMsg
</Blueprint3.Card>
`;
